File: update_images.py
import json

# Read the products.json file
with open('products.json', 'r', encoding='utf-8') as f:
    products = json.load(f)

# Image mapping based on category and style
def get_image_url(product):
    category = product.get('category', '')
    style = product.get('style', '')
    
    # Alines
    if category == 'Aline':
        return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Long%20Top%20Alines/Aline__Far.webp'
    
    # Capri
    if category == 'Capri':
        if style == 'Core':
            return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Core%20Capri/Capri__CHS.webp'
    
    if category == 'Fancy Capri':
        return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Fancy%20Capri/Capri_Denim%20_Cadd%2032%20no.webp'
    
    # Shorts
    if category == 'Shorts':
        if style == 'Core':
            return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Core%20Shorts/Bloomer__BFC.webp'
    
    if category == 'Denim Shorts':
        return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Fancy%20Shorts/Shorts_Denim_DSC3.webp'
    
    # Skirts
    if category == 'Skirts':
        if style == 'Core':
            return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Core%20Skirts/Skirt__Box%20Seemar.webp'
        elif style == 'Fancy':
            return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Fancy%20Skirts/Skirt_Denim_SDMS819.webp'
    
    if category == 'Denim Skirts':
        return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Fancy%20Skirts/Skirt_Denim_SDMS819.webp'
    
    # Tops
    if category == 'Tops(T-shirt)':
        if style == 'Fancy':
            return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Fancy%20Tops/Top__DT288.webp'
        # Note: No Core Tops in current data, but adding for completeness
        elif style == 'Core':
            return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Core%20Tops/Top__GT144.webp'
    
    if category == 'Denim Top':
        return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Fancy%20Tops/Top__DT288.webp'
    
    # Pants / Full Bottom
    if category == 'Denim Pants':
        return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Fancy%20Full%20Bottom/Pant_Denim_Denim%20Basic%20Jegging.webp'
    
    # Sets - using Core Tops as default for now
    if category == 'Set':
        if style == 'Core':
            return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Core%20Tops/Top__GT144.webp'
        elif style == 'Fancy':
            return 'https://tahoismsecifrazmevgm.supabase.co/storage/v1/object/public/products/Fancy%20Tops/Top__DT288.webp'
    
    # Default fallback (shouldn't reach here)
    return product.get('image_url', '')

# Update all products
updated_count = 0
for product in products:
    old_url = product.get('image_url', '')
    new_url = get_image_url(product)
    
    if new_url and new_url != old_url:
        product['image_url'] = new_url
        updated_count += 1
        print(f"Updated {product['design_name']} ({product['category']}, {product['style']})")

# Write back to file
with open('products.json', 'w', encoding='utf-8') as f:
    json.dump(products, f, indent=4, ensure_ascii=False)

print(f"\n✅ Successfully updated {updated_count} products!")
print("Image mapping complete!")

File: update_images.js
import { readFileSync, writeFileSync } from 'fs';

// Read both files
const products = JSON.parse(readFileSync('products.json', 'utf8'));
const imageMapping = JSON.parse(readFileSync('progress.txt', 'utf8'));

// Create a map of Sr. No. to Image URL
const imageMap = new Map();
imageMapping.forEach(item => {
    const srNo = item['Sr. No.'];
    let imageUrl = item['Images'] || '';

    // Clean up the URL (trim whitespace)
    imageUrl = imageUrl.trim();

    // Skip placeholder images
    if (imageUrl && !imageUrl.includes('placeholder')) {
        imageMap.set(srNo, imageUrl);
    }
});

// Update products with correct images
let updatedCount = 0;
let notFoundCount = 0;

products.forEach(product => {
    const srNo = product.sr_no;

    if (imageMap.has(srNo)) {
        const newUrl = imageMap.get(srNo);
        const oldUrl = product.image_url;

        if (newUrl !== oldUrl) {
            product.image_url = newUrl;
            updatedCount++;
            console.log(`✅ Updated #${srNo}: ${product.design_name}`);
        }
    } else {
        notFoundCount++;
        console.log(`⚠️  No image found for #${srNo}: ${product.design_name}`);
    }
});

// Write back to file
writeFileSync('products.json', JSON.stringify(products, null, 4), 'utf8');

console.log(`\n${'='.repeat(50)}`);
console.log(`✅ Successfully updated ${updatedCount} products!`);
console.log(`⚠️  ${notFoundCount} products have no specific image mapping`);
console.log(`${'='.repeat(50)}`);
console.log('\nImage mapping from progress.txt applied successfully!');
